design/FY2021/NFP-Spaces.md

```diff
@@ -0,0 +1,234 @@
+Enclosures are assigned automatically based on the zones connected by an air boundary surface. 
+
+
+## New Space Object
+
+### Relationship assumptions ##
+
+If we keep the current definition of "Zone" and add the concept of
+"Space" (<= Zone), then this would add a new layer to the data model:
+
+    Zone
+        Thermostat
+        HVAC equipment
+        Space 2
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+        Space 3
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+
+    Enclosure
+        Space 1
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+        Space 2
+            Surfaces
+            Internal Gains (People, Lights, etc.)
+    
+
+### Definitions
+
+ * Surface - A geometric plane which is attached to a Space. 
+   * A Surface can be opaque, transparent, or an air boundary.
+   * Each Surface belongs to one Space.
+
+ * Space - A collection of one or more Surfaces and internal gains.
+   * Each Space belongs to one Zone (explicitly user-assigned).
+   * The Spaces in a Zone are lumped together for the Zone heat balance.
+   * There is no heat balance at the Space level.
+   * Each Space belongs to one Enclosure (implicitly assigned).
+
+ * Zone - An air mass connecting Surfaces, internal gains, and HVAC equipment for heat balance and HVAC control.
+   * Each Zone is comprised of one or more Spaces.
+   * All Surfaces and internal gains associated with the Spaces are included in the Zone.
+   * The Zone heat balance does not change: it has an air node (or a Room Air Model) and includes all Surfaces and 
+   internal gains from its Spaces plus any HVAC which is attached to the Zone.
+
+ * Enclosure - A continuous volume connecting Surfaces for radiant, solar, and daylighting exchange.
+   * Each Enclosure is comprised of one or more Spaces.
+   * There is one Enclosure for each Space unless there are air boundary surfaces 
+   which group multiple Spaces into a single Enclosure.
+   * All Surfaces and internal radiant gains associated with the Spaces are included in the Enclosure.
+   * Enclosures only distribute radiant (thermal), solar, and visible energy to and from the Surfaces.
+   * There is no full heat balance at the Enclosure level. Each Enclosure only balances the radiant/solar flux on each 
+   Surface. These fluxes then become part of the Surface inside heat balance.
+
+## Proposed Input Approach ##
+Very minimal changes to current inputs.
+
+ * Old Zone object becomes Space object plus some tags.
+ * New Zone object has a name and a list of Spaces.
+ * Surfaces will reference a Space instead of a Zone.
+ * ZoneHVAC and Thermostats remain as-is and continue to reference a Zone or ZoneList.
+ * New SpaceList object (equivalent to ZoneList, but for Spaces).
+ * Internal gains reference Space or Spacelist (instead of Zone or ZoneList).
+ * ZoneInfiltration and ZoneVentilation remain at the zone level.
```

Comment:
Is it possible to make infiltration or ventilation at the space level?  That would simplify input for cases where ventilation requirements are a function of space type.

Reply:
I think it is possible, I think we probably just want to avoid having both (e.g. SpaceInfiltration *and* ZoneInfiltration). Moving everything to the space level will probably be more work, but if that's the right solution then we should do that.

Reply:
Whether it's SpaceInfiltration or ZoneInfiltration, it'd be nice to be able to get individual outputs for each object. With current EnergyPlus, you can have multiple ZoneInfiltration objects in a zone, but can only get aggregate outputs (i.e., the key for Zone Infiltration Sensible Heat Gain Energy is a zone object, not an infiltration object).

Reply:
Should probably have outputs for both space and zone to show individual as well as aggregated.

Reply:
I agree with this: limit to one input approach but output at both the space and zone level.